docs/doc/60-contributing/03-rfcs/20221105-infer-schema.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Infer Schema
+---
+
+- RFC PR: [datafuselabs/databend#8645](https://github.com/datafuselabs/databend/pull/8645)
+- Tracking Issue: [datafuselabs/databend#8646](https://github.com/datafuselabs/databend/issues/8646)
+
+## Summary
+
+Allowing infer schema from staged files or locations.
+
+## Motivation
+
+To load data from a stage or location, users need to create table first. However, sometimes users don't know the file schema, or the schema is too complex / too simple to be input by users.
+
+Allowing infer schema from existing files makes our users' lives easier. Also, this feature will unlock the databend from implementing `select * from @my_stage`.
+
+## Guide-level explanation
+
+Users will be able to infer the schema of the staged file by:
+
+```sql
+INFER @my_stage/data.csv FILE_FORMAT = ( TYPE = CSV);
```

Comment:
From the sql style, it would be better to use the table function as:
```
 select * from infer_schema(
      @my_stage/data.csv,
      FILE_FORMAT = ( TYPE = CSV)
      )
```

INFER is not a special type of SQL.

Reply:
And we can create the table as:
```
CREATE TABLE t as  select * from infer_schema(
      @my_stage/data.csv,
      FILE_FORMAT = ( TYPE = CSV)
      )
```

Reply:
> From the sql style, it would be better to use the table function as:

This table function will be complex. We must do hacks on our new expression and can't reuse the work from `COPY` / `CREATE STAGE`. For example, we don't support named args so far.

> And we can create the table as:

`LIKE`'s semantic is different.

```sql
CREATE TABLE t LIKE  select * from infer_schema(
      @my_stage/data.csv,
      FILE_FORMAT = ( TYPE = CSV)
      )
```

will create a table with schema `(COLUMN_NAME VARCHAR, TYPE VARCHAR, NULLABLE BOOL)`.

We need a new syntax. (Snowflake uses `USEING TEMPLATE ( <Query> )`)

Reply:
Make sense.